crates/biome_css_analyze/src/lint/nursery/use_sorted_properties.rs

```diff
@@ -0,0 +1,387 @@
+            ActionCategory::QuickFix(Cow::Borrowed("")),
+            ctx.metadata().applicability(),
+            markup! { "Sort these properties" }.to_owned(),
+            mutation,
+        ));
+    }
+}
+
+/// "Recess order" is a semantic ordering that mimics the behavior of [stylelint-config-recess-order](https://github.com/stormwarning/stylelint-config-recess-order).
+///
+/// Which in turn mimics the behavior of twitter's [RECESS](https://github.com/twitter-archive/recess/blob/29bccc870b7b4ccaa0a138e504caf608a6606b59/lib/lint/strict-property-order.js).
+pub struct RecessOrderProperties(pub BTreeSet<SortInfo>);
+
+impl RecessOrderProperties {
+    pub fn new(items: &[AnyCssDeclarationOrRule]) -> Self {
+        RecessOrderProperties(
+            items
+                .iter()
+                .enumerate()
+                .map(|(position, node)| {
+                    let mut sort_info = SortInfo::from(node);
+                    sort_info.original_position = position;
+                    sort_info
+                })
+                .collect::<BTreeSet<SortInfo>>(),
+        )
+    }
+
+    pub fn is_sorted(&self) -> bool {
+        // The list is sorted if the original_position field equals the actual position for every item
+        self.0
+            .iter()
+            .enumerate()
+            .all(|(position, item)| position == item.original_position)
+    }
+
+    pub fn as_sorted(
+        &self,
+        original_items: &[AnyCssDeclarationOrRule],
+    ) -> Option<CssDeclarationOrRuleList> {
+        CssDeclarationOrRuleList::cast(SyntaxNode::new_detached(
+            CssSyntaxKind::CSS_DECLARATION_OR_RULE_LIST,
+            self.0.iter().map(|sort_info| {
+                original_items
+                    .get(sort_info.original_position)
+                    .map(|node| NodeOrToken::Node(node.clone().into_syntax()))
+            }),
+        ))
+    }
+}
+
+fn css_identifier_to_prop_text(ident: &CssIdentifier) -> Option<TokenText> {
+    let tok = ident.value_token().ok()?;
+    Some(tok.token_text_trimmed())
+}
+
+fn css_declaration_to_prop_text(
+    decl_with_semicolon: &CssDeclarationWithSemicolon,
+) -> Option<TokenText> {
+    decl_with_semicolon
+        .declaration()
+        .ok()?
+        .property()
+        .ok()?
+        .as_css_generic_property()?
+        .name()
+        .ok()
+        .and_then(|prop_name| -> Option<TokenText> {
+            if let AnyCssDeclarationName::CssIdentifier(ident) = prop_name {
+                Some(ident.value_token().ok()?.token_text_trimmed())
+            } else {
+                None
+            }
+        })
+}
+
+/// Returns a declaration's vendor prefix in lowercase
+fn prop_text_to_prefix(tok_text: &TokenText) -> Option<&'static str> {
+    let prop_lowercase = tok_text.text().to_lowercase_cow();
+    let prefix = vender_prefix(&prop_lowercase);
+    if prefix.is_empty() {
+        return None;
+    }
+    Some(prefix)
+}
```

Comment:
We have a crate called `biome_string_case` to handle casing. Please use this crate, as it handles everything for you.

Reply:
I'm using `to_lowercase_cow` from `biome_string_case`, do you mean for it to use `Case::convert` instead?

Using `Case::convert` would allocate a lot of `String`. The current implementation only allocates when the string is not already lowercase.